crates/store/src/inner_forest.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::BTreeMap;
+
+use miden_objects::account::{AccountId, AccountStorage, StorageSlotContent};
+use miden_objects::block::BlockNumber;
+use miden_objects::crypto::merkle::SmtForest;
+use miden_objects::{EMPTY_WORD, Word};
+
+/// Container for forest-related state that needs to be updated atomically.
+pub(crate) struct InnerForest {
+    /// `SmtForest` for efficient account storage reconstruction.
+    /// Populated during block import with storage and vault SMTs.
+    pub(crate) storage_forest: SmtForest,
```

Comment:
nit: I would not prefix this with `storage_` to avoid a potential interpretation that this field is only for storage maps. Probably just `forest` should be fine.